src/codegen/object.rs

```diff
@@ -141,7 +141,22 @@ pub fn generate(
 
     if need_generate_inherent(analysis) {
         writeln!(w)?;
-        write!(w, "impl {} {{", analysis.name)?;
+        // We don't generate an impl block if the type doesn't have any of the followings:
+        // - Constructors / Functions / Builder properties (no build function)
+        // - Is a final type & doesn't have either methods / properties / child properties / signals
+        let should_generate_impl_block = !analysis.constructors().is_empty()
```

Comment:
That seems overly complicated... Is it really worth it?

Reply:
Well, it's complicated yes, but that's how things are :) Whether it's worth it or not; that's up to you as you opened the issue :p

Reply:
You're usually the one complaining about such things in autogenerated code :)

Reply:
I could move the logic to analysis and it would be just a func call here

Reply:
I moved things around so it looks "simpler"